lte/gateway/c/core/oai/tasks/amf/amf_recv.cpp

```diff
@@ -399,6 +401,52 @@ int amf_handle_registration_request(
             found_imsi->second = guti_and_amf_id;
           }
         }
+      } else {
+        /*
+         * Extract the SUPI or IMSI from SUCI as scheme output is not encrypted
+         */
+        // params->imsi = new imsi_t();
+        /* Copying PLMN to local supi which is imsi*/
+        supi_imsi.plmn.mcc_digit1 =
+            msg->m5gs_mobile_identity.mobile_identity.imsi.mcc_digit1;
+        supi_imsi.plmn.mcc_digit2 =
+            msg->m5gs_mobile_identity.mobile_identity.imsi.mcc_digit2;
+        supi_imsi.plmn.mcc_digit3 =
+            msg->m5gs_mobile_identity.mobile_identity.imsi.mcc_digit3;
+        supi_imsi.plmn.mnc_digit1 =
+            msg->m5gs_mobile_identity.mobile_identity.imsi.mnc_digit1;
+        supi_imsi.plmn.mnc_digit2 =
+            msg->m5gs_mobile_identity.mobile_identity.imsi.mnc_digit2;
+        supi_imsi.plmn.mnc_digit3 =
+            msg->m5gs_mobile_identity.mobile_identity.imsi.mnc_digit3;
+
+        ue_context->amf_context.m5_guti.guamfi.plmn.mcc_digit1 =
+            supi_imsi.plmn.mcc_digit1;
+        ue_context->amf_context.m5_guti.guamfi.plmn.mcc_digit2 =
+            supi_imsi.plmn.mcc_digit2;
+        ue_context->amf_context.m5_guti.guamfi.plmn.mcc_digit3 =
+            supi_imsi.plmn.mcc_digit3;
+        ue_context->amf_context.m5_guti.guamfi.plmn.mnc_digit1 =
+            supi_imsi.plmn.mnc_digit1;
+        ue_context->amf_context.m5_guti.guamfi.plmn.mnc_digit2 =
+            supi_imsi.plmn.mnc_digit2;
+        ue_context->amf_context.m5_guti.guamfi.plmn.mnc_digit3 =
+            supi_imsi.plmn.mnc_digit3;
+
+        ue_context->amf_context.reg_id_type = M5GSMobileIdentityMsg_SUCI_IMSI;
+
+        std::string empheral_public_key = reinterpret_cast<char*>(
+            msg->m5gs_mobile_identity.mobile_identity.imsi.empheral_public_key);
+        std::string ciphertext = reinterpret_cast<char*>(
+            msg->m5gs_mobile_identity.mobile_identity.imsi.ciphertext);
+        std::string mac_tag = reinterpret_cast<char*>(
+            msg->m5gs_mobile_identity.mobile_identity.imsi.mac_tag);
+
+        get_decrypt_imsi_suci_extension(
+            &ue_context->amf_context,
+            msg->m5gs_mobile_identity.mobile_identity.imsi.home_nw_id,
+            empheral_public_key, ciphertext, mac_tag);
+        OAILOG_FUNC_RETURN(LOG_NAS_AMF, rc);
```

Comment:
is this blob covered by any unit tests?

Reply:
Here part of this blob that is Decrypt SUCI request message to Subscriberdb and the Decrypted SUCI response message is validated using UT code.

Reply:
you need to add unit test coverage for this blob. (cc @panyogesh)

Reply:
Optimized this code and added the UT for the optimized code. For function get_decrypt_imsi_suci_extension, it involves GRPC call to subscriberdb, so have written UT for messages M5GSUCIRegistrationRequest and M5GSUCIRegistrationAnswer separately.